### PR TITLE
chore: replace retention date picker

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

Towards https://github.com/PostHog/posthog/issues/13624

## Changes

Replace the `RetentionDatePicker`

Because retention only needs an hour selector and not minutes I've changed the `showTime` prop to specify `granularity` instead

|Before|After|
|----|----|
| Screenshot 2024-05-21 at 10 39 16 | Screenshot 2024-05-21 at 10 30 03 |

|Before|After|
|----|----|
| Screenshot 2024-05-21 at 10 39 31
 | Screenshot 2024-05-21 at 10 38 32 |

### CODE REVIEW

File: frontend/__snapshots__/scenes-app-insights--retention-breakdown-edit--light.png
Comment (benjackwhite) [inline frontend/__snapshots__/scenes-app-insights--retention-breakdown-edit--light.png:1]: I _wonder_ if the button should be a secondary type?

File: frontend/src/lib/lemon-ui/LemonCalendar/LemonCalendar.tsx [hunk 1]
```diff
@@ -25,8 +25,8 @@ export interface LemonCalendarProps {
     months?: number
     /** 0 or unset for Sunday, 1 for Monday. */
     weekStartDay?: number
-    /** Show a time picker */
-    showTime?: boolean
+    /** Set the time granularity of the calendar */
+    granularity?: 'day' | 'hour' | 'minute'
```
Comment (benjackwhite) [inline frontend/src/lib/lemon-ui/LemonCalendar/LemonCalendar.tsx:29]: nice!

**Review by benjackwhite — APPROVED** [review]
Didn't try it but it all looked pretty good to me

File: frontend/__snapshots__/scenes-app-insights--retention-breakdown-edit--light.png
Comment (daibhin) [reply frontend/__snapshots__/scenes-app-insights--retention-breakdown-edit--light.png:1]: I'll leave that up to @PostHog/team-product-analytics to decide 😅

**Pull request merged**